R/derive_vars_extreme_event.R

```diff
@@ -0,0 +1,228 @@
+#' Add the Worst or Best Observation for Each By Group as New Variables
+#'
+#' Add the first available record from `events` for each by group as new
+#' variables, all variables of the selected observation are kept. It can be used
+#' for selecting the extreme observation from a series of user-defined events.
+#'
+#' @param events Conditions and new values defining events
+#'
+#'   A list of `event()` or `event_joined()` objects is expected. Only
+#'   observations listed in the `events` are considered for deriving extreme
+#'   event. If multiple records meet the filter `condition`, take the first
+#'   record sorted by `order`. The data is grouped by `by_vars`, i.e., summary
+#'   functions like `all()` or `any()` can be used in `condition`.
+#'
+#'   For `event_joined()` events the observations are selected by calling
+#'   `filter_joined()`. The `condition` field is passed to the `filter` argument.
```

Comment:
Should be `filter_join` instead of `filter` as `filter` is deprecated.

Reply:
Updated